e2e/authorization_test.go

```diff
@@ -16,149 +16,258 @@ package e2e
 
 import (
 	"context"
-	"errors"
 	"fmt"
 	"strings"
 	"testing"
-	"time"
 
 	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/kubeutil"
 	"github.com/GoogleCloudPlatform/prometheus-engine/e2e/operatorutil"
 	"github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator"
 	monitoringv1 "github.com/GoogleCloudPlatform/prometheus-engine/pkg/operator/apis/monitoring/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
-	"k8s.io/apimachinery/pkg/util/wait"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-func TestTLS(t *testing.T) {
-	t.Parallel()
-	tctx := newOperatorContext(t)
-	ctx := context.Background()
+func isPodMonitoringTargetCertificateError(message string) error {
+	const expected = "tls: failed to verify certificate: x509: certificate signed by unknown authority"
```

Comment:
Looks a bit brittle, but little harm in trying this! 😄 🤞🏽 

<img src="https://media2.giphy.com/media/kZnXc7VsUIaFWZvyQV/giphy.gif"/>

Reply:
Switched to the raw errors. Sadly it's coming from Prometheus' `/target` endpoint so we can't use `errors.Is`, but we can use string comparison.